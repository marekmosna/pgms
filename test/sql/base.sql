\set ECHO none
BEGIN;

\i test/pgtap-core.sql
\i sql/pgms.sql

SELECT plan(26);

SELECT has_type('spectrum');
---SELECT has_type('spectrumrange');

SELECT has_function('load_from_mgf');
SELECT has_function('load_from_mgf', ARRAY['character varying']);
SELECT has_function('load_from_mgf', ARRAY['oid']);
SELECT function_returns('load_from_mgf', 'setof record');

SELECT has_function('spectrum_normalize');
SELECT has_function('spectrum_normalize', ARRAY['spectrum']);
SELECT function_returns('spectrum_normalize', 'spectrum');

SELECT has_function('cosine_greedy');
SELECT has_function('cosine_greedy', ARRAY['spectrum', 'spectrum', 'real', 'real', 'real']);
SELECT function_returns('cosine_greedy', 'real');

SELECT has_function('spectrum_range');
SELECT has_function('spectrum_range', ARRAY['spectrum', 'character varying']);
SELECT function_returns('spectrum_range', 'spectrumrange');

SELECT has_function('load_from_json');
SELECT has_function('load_from_json', ARRAY['jsonb']);
SELECT function_returns('load_from_json', 'setof record');

SELECT has_function('cosine_modified');
SELECT has_function('cosine_modified', ARRAY['spectrum', 'spectrum', 'real', 'real', 'real', 'real']);
SELECT function_returns('cosine_modified', 'real');

SELECT has_function('intersect_mz');
SELECT has_function('intersect_mz', ARRAY['spectrum', 'spectrum', 'real']);
SELECT function_returns('intersect_mz', 'real');

SELECT has_function('precurzor_mz_match');
SELECT has_function('precurzor_mz_match', ARRAY['real', 'real', 'real', 'character varying']);
SELECT function_returns('precurzor_mz_match', 'real');

SELECT * FROM finish();
ROLLBACK;
