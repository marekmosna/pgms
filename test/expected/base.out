\set ECHO none
1..31
ok 1 - Type spectrum should exist
ok 2 - Function load_from_mgf() should exist
ok 3 - Function load_from_mgf(character varying) should exist
ok 4 - Function load_from_mgf(oid) should exist
ok 5 - Function load_from_mgf() should return setof record
ok 6 - Function spectrum_normalize() should exist
ok 7 - Function spectrum_normalize(spectrum) should exist
ok 8 - Function spectrum_normalize() should return spectrum
ok 9 - Function cosine_greedy() should exist
ok 10 - Function cosine_greedy(spectrum, spectrum, real, real, real) should exist
ok 11 - Function cosine_greedy() should return real
ok 12 - Function spectrum_range() should exist
ok 13 - Function spectrum_range(spectrum, character varying) should exist
ok 14 - Function spectrum_range() should return spectrumrange
ok 15 - Function load_from_json() should exist
ok 16 - Function load_from_json(jsonb) should exist
ok 17 - Function load_from_json() should return setof record
ok 18 - Function cosine_modified() should exist
ok 19 - Function cosine_modified(spectrum, spectrum, real, real, real, real) should exist
ok 20 - Function cosine_modified() should return real
ok 21 - Function intersect_mz() should exist
ok 22 - Function intersect_mz(spectrum, spectrum, real) should exist
ok 23 - Function intersect_mz() should return real
ok 24 - Function precurzor_mz_match() should exist
ok 25 - Function precurzor_mz_match(real, real, real, character varying) should exist
ok 26 - Function precurzor_mz_match() should return real
ok 27 - Function precurzor_mz_match(real[], real[], real, character varying, boolean) should exist
ok 28 - Function precurzor_mz_match(real[], real[], real, character varying, boolean) should return real[]
ok 29 - Function cosine_neutral_losses() should exist
ok 30 - Function cosine_neutral_losses(spectrum, spectrum, real, real, real, real, real) should exist
ok 31 - Function cosine_neutral_losses() should return real
